### PR TITLE
feat(error_reporting)!: Use new generated client classes

### PR DESCRIPTION
**Pull request by dazuma**

### CODE REVIEW

File: google-cloud-error_reporting/lib/google/cloud/error_reporting.rb [hunk 1]
```diff
@@ -62,8 +62,7 @@ module ErrorReporting
       #   * `https://www.googleapis.com/auth/cloud-platform`
       #
       # @param [Integer] timeout Default timeout to use in requests. Optional.
-      # @param [Hash] client_config A hash of values to override the default
-      #   behavior of the API client. Optional.
+      # @param [Hash] client_config Unused.
```
Comment (quartzmo) [inline google-cloud-error_reporting/lib/google/cloud/error_reporting.rb:65]: Do you want to explicitly label this as deprecated?

Comment (dazuma) [reply google-cloud-error_reporting/lib/google/cloud/error_reporting.rb:65]: Yeah, that's a good idea, we should.

Comment (dazuma) [reply google-cloud-error_reporting/lib/google/cloud/error_reporting.rb]: Done

File: google-cloud-error_reporting/lib/google-cloud-error_reporting.rb [hunk 2]
```diff
@@ -89,8 +88,7 @@ def error_reporting scope: nil, timeout: nil, client_config: nil
     #   * `https://www.googleapis.com/auth/cloud-platform`
     #
     # @param [Integer] timeout Default timeout to use in requests. Optional.
-    # @param [Hash] client_config A hash of values to override the default
-    #   behavior of the API client. Optional.
+    # @param [Hash] client_config Unused.
```
Comment (quartzmo) [inline google-cloud-error_reporting/lib/google-cloud-error_reporting.rb:91]: Also: Unused and deprecated.

Comment (dazuma) [reply google-cloud-error_reporting/lib/google-cloud-error_reporting.rb]: Fixed.

**Pull request merged**